src/dsis_client/api/client/dsis_client.py

```
@@ -204,3 +124,51 @@ def cast_results(self, results: List[Dict[str, Any]], schema_class) -> List[Any]
             >>> faults = client.cast_results(response["value"], Fault)
         """
         return cast_results(results, schema_class)
+
+    def _aggregate_nextlink_pages(
+        self, response: Dict[str, Any], endpoint: str
+    ) -> Dict[str, Any]:
+        """Follow OData nextLink(s) and aggregate paged results into a single response.
```

little concerned about memory.  maybe here should be a yield so only one page in memory, we let the caller to handle the aggregation. simply put, better to let it crash on user side, not package side.